src/NRules/NRules/Rete/AggregateNode.cs

```diff
@@ -30,7 +30,7 @@ internal class AggregateNode : BinaryBetaNode
         _isSubnetJoin = isSubnetJoin;
     }
 
-    public override void PropagateAssert(IExecutionContext context, List<Tuple> tuples)
+    public override void PropagateAssert(IExecutionContext context, IReadOnlyCollection<Tuple> tuples)
```

Comment:
@gritcsenko thanks for rescinding this PR. Need to be very careful with some of these changes. For example, enumerating a List via an interface causes boxing/unboxing of the enumerator struct, which is pretty bad for performance.
So, I would like to avoid these types of blanket changes of going and replacing all concrete collections with interfaces